collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ActivityGeoDataRequester.kt

```diff
@@ -0,0 +1,164 @@
+package org.odk.collect.android.widgets.utilities
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import org.javarosa.form.api.FormEntryPrompt
+import org.odk.collect.android.utilities.Appearances
+import org.odk.collect.android.utilities.ApplicationConstants
+import org.odk.collect.android.utilities.FormEntryPromptUtils
+import org.odk.collect.android.widgets.interfaces.GeoDataRequester
+import org.odk.collect.geo.Constants.EXTRA_DRAGGABLE_ONLY
+import org.odk.collect.geo.Constants.EXTRA_READ_ONLY
+import org.odk.collect.geo.Constants.EXTRA_RETAIN_MOCK_ACCURACY
+import org.odk.collect.geo.GeoPointActivity
+import org.odk.collect.geo.GeoPointMapActivity
+import org.odk.collect.geo.GeoPolyActivity
+import org.odk.collect.permissions.PermissionListener
+import org.odk.collect.permissions.PermissionsProvider
+import java.lang.Boolean.parseBoolean
+
+class ActivityGeoDataRequester(
```

Comment:
I looked at converting the test as well but backed out when I didn't get a clean auto convert - don't want to hold this up or introduce more risk than we need to.